devices/At24cxx/At24Base.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.At24cxx
+{
+    /// <summary>
+    /// Base class for common functionality of the At24c serices of devices.
+    /// </summary>
+    public abstract class At24Base : IDisposable
+    {
+        /// <summary>
+        /// Default I2C address for At24cxx familly.
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x50; // 01010000
+
+        /// <summary>
+        /// Underlying I2C device.
+        /// </summary>
+        private readonly I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Gets number of pages available on the device.
+        /// </summary>
+        public int PageCount { get; private set; }
+
+        /// <summary>
+        /// Gets device page size (in bytes).
+        /// </summary>
+        public int PageSize { get; private set; }
+
+        /// <summary>
+        /// Gets available memory on the device (in bytes).
+        /// </summary>
+        public int Size { get; private set; }
+
+        /// <summary>
+        /// Helper function to create a buffer with the first two bytes already populated by the device address.
+        /// </summary>
+        /// <param name="address">The device address.</param>
+        /// <param name="length">The length of data the buffer is required to hold.</param>
+        /// <returns>A buffer with the first two bytes populated with the device address.</returns>
+        private static byte[] CreateWriteBuffer(int address, int length)
```

Comment:
why not having address as a byte, will avoid the cast as byte later on

Reply:
It depends on the chip, all devices with over 16kb of memory expect a 16 bit memory address written as a high and low byte that needs to be sent before the data to write.

I dropped support for chips in the family with shorter memory addresses since some bits from the memory address were tacked on to the device address each time data was written and it didn't play well with how I2cDevice currently works.